external-crates/move/crates/move-compiler/src/typing/core.rs

```diff
@@ -2191,6 +2231,16 @@ pub fn solve_constraints(context: &mut Context) {
             _ => (),
         }
     }
+
+    let divergent_vars = subst.divergent_vars.clone();
+    for (divergent_var, loc) in divergent_vars {
+        let last_tvar = forward_tvar(&subst, divergent_var);
+        if subst.get(last_tvar).is_none() {
+            join_bind_tvar(&mut subst, loc, last_tvar, sp(loc, Type_::Unit))
```

Comment:
This feels... strange to me. Maybe we should add void, but just here! lol
Thoughts?